src/vs/workbench/contrib/terminalContrib/chatAgentTools/browser/task/taskHelpers.ts

```diff
@@ -66,7 +67,8 @@ export async function getTaskForTool(id: string | undefined, taskDefinition: { t
 
 	let tasksForWorkspace;
 	for (const [folder, tasks] of workspaceFolderToTaskMap) {
-		if (URI.parse(folder).path === workspaceFolder) {
+		// Use isEqual to compare URIs for cross-platform compatibility
+		if (isEqual(URI.parse(folder), URI.file(workspaceFolder))) {
```

Comment:
@meganrogge is this a safe use of `URI.file`? What if this is running from web with a remote URI?

Reply:
yes there was this followup https://github.com/microsoft/vscode/pull/259231/files